Packages/com.unity.inputsystem/Tests/InputSystem/CoreTests_Layouts.cs

```diff
@@ -1853,6 +1853,63 @@ public void Layouts_CanSetUpMultipleLayoutsFromSingleState_UsingVariants()
         Assert.That(deviceB["axis"].variants, Is.EqualTo("B"));
     }
 
+    [Test]
+    [Category("Layouts")]
+    public void Layouts_ParseParameters_EmptyStringReturnsNull()
+    {
+        var parameters = InputControlLayout.ParseParameters("");
+        Assert.AreEqual(null, parameters, "Empty string should generate null parameters");
+    }
+
+    void Layouts_ParseParameters_GenericWorks<T>(T[] expectedValues, InputControlLayout.ParameterType parameterType, string input) where T : unmanaged
+    {
+        var parameters = InputControlLayout.ParseParameters(input);
+        Assert.AreEqual(expectedValues.Length, parameters.Length, "Number of parameters does not match.");
+        unsafe
+        {
+            for (int i = 0; i < expectedValues.Length; i++)
+            {
+                var p = parameters[i];
+                Assert.AreEqual(parameterType, p.type, $"Unexpected type in parameter {i}");
+                Assert.AreEqual($"p{i}", p.name, $"Unexpected name in parameter {i}");
+                Assert.AreEqual(expectedValues[i], *(T*)p.value, $"Unexpected value in parameter {i}");
+            }
+        }
+    }
```

Comment:
Remove helper and replace the tests with a single test that takes all the test permutations as TestCase attributes.

Reply:
I don't think that works with the generic argument.